### ISSUE TITLE
Giving TestCaseAttribute an explicit decimal for a parameter that is a decimal gives a Nunit 1001 error

### ISSUE THREAD

**Issue by Hounsvad** [nunit/nunit.analyzers#509]
I am working with nunit 3.13.3

This is the offending case
```
    [TestCase(3650,-0.000027m)]
    public void Test_CalculateDegradationRate(int numberOfCycles, decimal expectedDegradationRate)
    {
        Assert.That(WithinTolerance(_state.DegradationRate, expectedDegradationRate), Is.True);
    }
```
It gives the error if the decimal is explicitly declared as a decimal with either 'm' or 'M', but it works fine if no specifier is given or if a 'd' specifier is given.

![image](https://user-images.githubusercontent.com/36563020/214784121-49457053-73d3-4402-9913-41160619c93b.png)

I would of course expect that the TestCaseAttribute would at least take an explicit decimal.

**Comment by Hounsvad** [comment]
Left the issue on the wrong repo

**Comment by manfred-brands** [comment]
This is the correct repo as the analyzer raises this error.

**Comment by mikkelbu** [comment]
I didn't see this first, so I've moved the other issue - #510 - back 😄. I cannot remember the code, but I think the diagnostic is raise as the code does not compile.
